src/lib/cache/gdfs_cache.hpp

```diff
@@ -30,10 +34,11 @@ class GDFSCache : public AbstractCacheImpl<Key, Value> {
 
   using Handle = typename boost::heap::fibonacci_heap<GDFSCacheEntry>::handle_type;
   using CacheMap = typename std::unordered_map<Key, Handle>;
+  using CacheSnapshot = typename std::unordered_map<Key, GDFSCacheEntry>;
```

Comment:
Why is this defined here, not in AbstractCache?

Reply:
Probably related: can we put `Key` and `Value` into the snapshot? Or is there a particular reason to exclude them? (receiving frequency might be one, now that I think about it while writing ...)

I would prefer a solution to use the snapshot that does not require knowledge about the GDFS internals.

Reply:
Ah, I didn't even see that the value was GDFSCacheEntry. I guess this only depends on whether you are interested in the frequency values or not. Apparently, you just want to know *what* is in the cache, not *why*.

Reply:
Okay, I did not know if that information was useful. So I'll change it to an `unordered_map<Key, Value>`.

Reply:
But that leaves out the frequency.

Reply:
Right :D
So I guess something like 
```c++
struct SnapshotEntry {
    Value value;
    size_t frequency;
};
...
std::unordered_map<Key, SnapshotEntry> snapshot() {...}
```
would be fine?

Reply:
> But then it again is the question of whether to define it in the abstract class or not, as we might assume any future implementations to deal with the frequency.

Correct. That shouldn't go into AbstractCache.

@Bouncner and @Bensk1 should decide - they are the ones using this.

Reply:
> I would prefer a solution to use the snapshot that does not require knowledge about the GDFS internals.

Either you want the frequency or you don't... ;)

Reply:
I want the frequency, but I don't care if it's a GDFS or not. A plan cache without a frequency is of really small value to me, but I really don't care if there are changes to the caching at some point in time. When a new caching algorithm does not provide a frequency out of the box, I think it needs to be added or this caching algorithms is simply not a good fit for a database system.

Actually, I'd love even more like HANAs "time of adding entry to cache", but that's another story.

It's hard for me to follow this whole thread here. So sorry if that has been already discussed.

+1 for the SnapshotEntry.

Reply:
The SnapshotEntry doesn't make any sense. It pretends to be a common interface for this and other (non-existent) implementations, but those do not necessarily have a frequency field (e.g., LRU).

If you want the snapshot to contain the frequency, I consider the current approach to be correct.

Reply:
I think it makes sense as it states what we expect of the underlying cache. If the LRU cannot provide it: LRU caches don't fit.
Or do I oversee components in Hyrise that also use a cache and where the frequency does not make any sense?

From a cache user perspective I think having manually get the frequency from some objects types that I don't understand is cumbersome.

Reply:
For the sake of convenience, I could live with an `optional<..> frequency`. Still don't think that it's architecturally correct.